stdlib/public/core/StringComparable.swift

```diff
@@ -80,33 +80,25 @@ extension _StringGuts {
   @effects(readonly)
   public
   static func _compareDeterministicUnicodeCollation(
-    _leftUnsafeStringGutsBitPattern leftBits: (UInt, UInt),
-    _rightUnsafeStringGutsBitPattern rightBits: (UInt, UInt)
+    _leftUnsafeStringGutsBitPattern leftBits: (UInt64, UInt64),
```

Comment:
Err, since this might get very messy very quick (depending on how the 32-bit form takes shape) how about a type alias:
```swift
extension StringGuts {
#if ...
  typealias _RawBitPattern = (UInt, UInt)
# else if ...
  typealias _RawBitPattern = ...
...
}
```